larq/optimizers_v2.py

```diff
@@ -5,6 +5,84 @@
 from copy import deepcopy
 
 
+@utils.register_keras_custom_object
+class OptimizerGroup(tf.keras.optimizers.Optimizer):
+    """Group of a full-precision and binary optimizer.
+
+    #TODO: Do we want to abstract this to `n` optimizers, not specifically one fp and
+    one bin? (I personally prefer the user-side simplicity of this current approach.)
```

Comment:
I'm strongly in favour of the more general approach; I don't see a huge benefit in making this so specific while it removes a lot of flexibility.

Reply:
(See discussion [here](https://github.com/larq/larq/pull/304/files/4ce2e10b4c5ecbffc5f9ec1353f7cf5ee0a625d7#r341977897).)